client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useDispatch } from 'react-redux';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	usePromoCard: boolean;
+	source: 'email-comparison' | 'email-management';
+};
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildQueryString = ( parameters = {} ) =>
+		parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = encodeURIComponent(
+			emailManagementPurchaseNewEmailAccount(
+				selectedSite?.slug ?? '',
+				domain?.name ?? '',
+				null,
+				'login-redirect'
+			)
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const contactOwnerUrl = `https://privatewho.is/${ buildQueryString( {
+		s: domain?.name,
+	} ) }`;
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'owner_contact' | 'user_login' | 'contact_support' ) => {
+		const properties = {
+			owner_login: ownerUserName,
+		};
+
+		if ( eventType === 'owner_contact' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_owner_contact_click`, properties ) );
+
+			return;
+		} else if ( eventType === 'contact_support' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_contact_support_click`, properties ) );
+
+			return;
+		}
+
+		dispatch( recordTracksEvent( `calypso_email_providers_user_login_click`, properties ) );
```

Comment:
```suggestion
		if ( eventType === 'owner_contact' ) {
			dispatch( recordTracksEvent( `calypso_email_providers_owner_contact_click`, properties ) );
		} else if ( eventType === 'contact_support' ) {
			dispatch( recordTracksEvent( `calypso_email_providers_contact_support_click`, properties ) );
		} else if ( eventType === 'user_login' ) {
			dispatch( recordTracksEvent( `calypso_email_providers_user_login_click`, properties ) );
		}
```
Nitpicking, but I think this reads better. I'm guessing that the return statements are there because ESLint maybe doesn't like final `else` statements? In any case, it seems like we could do without the `return` statements here.